protocol/packages/swap/src/osmosis.rs

```diff
@@ -73,7 +83,41 @@ impl ExactAmountIn for Impl {
 
     // #[cfg(any(test, feature = "testing"))] revert TODO report a cargo bug that 'test' cfg is not applied
     #[cfg(feature = "testing")]
-    fn build_resp(amount_out: Amount) -> Any {
+    fn parse_request<GIn>(request: Any) -> SwapRequest<GIn>
+    where
+        GIn: Group,
+    {
+        let RequestMsg {
+            sender: _,
+            routes,
+            token_in: Some(token_in),
+            token_out_min_amount,
+        } = parse_request_from_any_and_type_url(request, RequestMsg::TYPE_URL)
+        else {
+            crate::pattern_match_else(type_name::<RequestMsg>())
+        };
+
+        assert_eq!({ token_out_min_amount }, "1");
```

Comment:
what is the purpose of this extra block?

Reply:
Nothing else than to force the value to be moved out and dropped on the spot after the comparison. Most likely I did that unintentionally as there isn't a loop or a long-running function afterwards.